cmds/task/actors.go

```diff
@@ -41,6 +43,54 @@ func runRerun(credentials *tcclient.Credentials, args []string, out io.Writer, _
 	return nil
 }
 
+// runRetrigger re-triggers a given task.
+// It will generate a new taskId, update timestamps and retries to 0
+func runRetrigger(credentials *tcclient.Credentials, args []string, out io.Writer, _ *pflag.FlagSet) error {
+	q := makeQueue(credentials)
+	taskID := args[0]
+
+	t, err := q.Task(taskID)
+	if err != nil {
+		return fmt.Errorf("could not get the task %s: %v", taskID, err)
+	}
+
+	newTaskID := slugid.V4()
+	now = time.Now().UTC()
+
+	newT := &queue.TaskDefinitionRequest{
+		SchedulerID: t.SchedulerID,
+		Created:  tcclient.Time(now),
+		Deadline: tcclient.Time(now.Add(24 * time.Hour)),
+		Expires:  tcclient.Time(now.Add(24*time.Hour).AddDate(0, 1, 0)),
+	}
+
+	// TaskDefinitionRequest: https://github.com/taskcluster/taskcluster-client-go/blob/88cfe471bfe2eb8fc9bc22d9cde6a65e74a9f3e5/tcqueue/types.go#L1368-L1549
+	// TaskDefinitionResponse: https://github.com/taskcluster/taskcluster-client-go/blob/88cfe471bfe2eb8fc9bc22d9cde6a65e74a9f3e5/tcqueue/types.go#L1554-L1716
+	newT.Dependencies  = t.Dependencies
+	newT.Extra         = t.Extra
+	newT.Metadata      = t.Metadata
+	newT.Payload       = t.Payload
+	newT.Priority      = t.Priority
+	newT.ProvisionerID = t.ProvisionerID
+	newT.Requires      = t.Requires
+	newT.Retries       = t.Retries
+	newT.Routes        = t.Routes
+	newT.SchedulerID   = t.SchedulerID
+	newT.Scopes        = t.Scopes
+	newT.Tags          = t.Tags
```

Comment:
Take a look at how it's done in taskcluster-tools...

I'm pretty sure there is an informal convention for adding either a tag or task.extra property pointing to the taskId it was retriggered from...